components/locale/src/canonicalizer.rs

```diff
@@ -37,11 +37,11 @@ use tinystr::TinyAsciiStr;
 ///
 /// [UTS #35: Annex C, LocaleId Canonicalization]: http://unicode.org/reports/tr35/#LocaleId_Canonicalization
 #[derive(Debug)]
-pub struct LocaleCanonicalizer {
+pub struct LocaleCanonicalizer<Expander = LocaleExpander> {
     /// Data to support canonicalization.
     aliases: DataPayload<AliasesV2Marker>,
     /// Likely subtags implementation for delegation.
-    expander: LocaleExpander,
+    expander: Expander,
```

Comment:
convention is to use single letters for generics, this looks like there's an `Expander` type. Can you use `E`?